### PR TITLE
Add wrong network modal to all templates

### PR DESCRIPTION
**Pull request by 0xaptosj**
Screenshot 2024-09-09 at 3 56 33 PM

### CODE REVIEW

File: templates/boilerplate-template/frontend/components/WrongNetwork.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    <Dialog.Root open={true}>
+      <Dialog.Portal>
+        <Dialog.Overlay className="fixed inset-0 z-50 bg-black/60 backdrop-blur-sm transition-opacity duration-300 ease-in-out" />
+        <Dialog.Content className="fixed z-50 top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2 shadow-lg rounded-lg p-6 bg-white dark:bg-gray-800 transition-transform duration-300 ease-in-out w-[90vw] max-w-md">
+          <div className="text-center">
+            <Dialog.Title className="text-3xl font-semibold mb-4 text-gray-900 dark:text-gray-100">
+              Wrong Network
+            </Dialog.Title>
+            <Dialog.Description className="text-lg text-gray-700 dark:text-gray-300">
+              You are currently on <span className="font-bold">{network?.name}</span>. Please switch to{" "}
```
Comment (0xmaayan) [inline templates/boilerplate-template/frontend/components/WrongNetwork.tsx:20]:
```suggestion
              Your wallet is currently on <span className="font-bold">{network?.name}</span>. Please switch to{" "}
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,8 @@ All notable changes to the create-aptos-dapp tool will be captured in this file.
 
 - Support Nextjs framework for Boilerplate template
 - Add new telegram mini app templates, we have 2 options, one using Aptos Wallet Adapter for explicit signing like other dapps, and the other using Mizu Wallet SDK Core for seamless signing experience.
+- Add a wrong network component to all templates to show when the user is on the wrong network
+- Add 2 new examples, `aptogotchi-keyless` and `aptogotchi-random-mint`
```
Comment (0xmaayan) [inline CHANGELOG.md:10]: remove?
```suggestion
```

File: templates/clicker-game-tg-mini-app-template/frontend/components/WrongNetwork.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { NETWORK } from "@/constants";
+import { useWallet } from "@aptos-labs/wallet-adapter-react";
+import * as Dialog from "@radix-ui/react-dialog";
+
+export function WrongNetwork() {
```
Comment (0xmaayan) [inline templates/clicker-game-tg-mini-app-template/frontend/components/WrongNetwork.tsx:5]:
```suggestion
export function WrongNetworkAlert() {
```

**Review by 0xmaayan — APPROVED** [review]
Nice! thank you for working on it.

File: CHANGELOG.md (hunk 2 above)
Comment (0xaptosj) [reply CHANGELOG.md:10]: i forgot to add that in the example PR

**Pull request merged**